aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -117,15 +117,12 @@ public void initialize(Map<String, String> properties) {
     }
 
     // Report Hadoop metrics if Hadoop is available
-    try {
-      DynConstructors.Ctor<MetricsContext> ctor =
-          DynConstructors.builder(MetricsContext.class).hiddenImpl(DEFAULT_METRICS_IMPL, String.class).buildChecked();
-      this.metrics = ctor.newInstance("s3");
-
-      metrics.initialize(properties);
-    } catch (NoSuchMethodException | ClassCastException e) {
-      LOG.warn("Unable to load metrics class: '{}', falling back to null metrics", DEFAULT_METRICS_IMPL, e);
-    }
+    String metricsImpl = properties.getOrDefault(
+            CatalogProperties.IO_METRICS_IMPL, CatalogProperties.IO_METRICS_IMPL_DEFAULT);
+    properties.put(CatalogProperties.IO_METRICS_SCHEME, properties.getOrDefault(
```

Comment:
we cannot guarantee the properties map is mutable. we should create a new copy to put the metrics scheme